w09h01/test/pgdp/pingutrip/TestreadWayPoints.java

```diff
@@ -0,0 +1,83 @@
+package pgdp.pingutrip;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static pgdp.pingutrip.PinguTrip.readWayPoints;
+
+public class TestreadWayPoints {
+    @Test
+    @DisplayName("No end (---)")
+    void test1() {
+        Stream<WayPoint> wayPointStream = readWayPoints("test/pgdp/pingutrip/path1.txt");
+
+        WayPoint[] expected = {new WayPoint(9.11, 1.1), new WayPoint(9.11, 1.1)};
+        WayPoint[] actual = wayPointStream.toArray(WayPoint[]::new);
+
+        assertEquals(expected.length, actual.length);
+
+        for(int i = 0; i < expected.length; i++) {
+            assertEquals(expected[i], actual[i], "The value at index " + i + " is wrong");
+        }
```

Comment:
I know, the existing the in `ReadWayPointsTest.java` is also not using `assertArrayEquals` but you use `assertArrayEquals` instead of the `assertEquals`?

```suggestion
        assertArrayEquals(expected, actual);
```